update composer.json for php-8-nightly compat

Kalloritis opened the pull request:
currently failing checks due to lack of regex match for nightly's of PHP (currently 8.0.0).

Kalloritis committed:
update composer.json for php-8-nightly compat

currently failing checks due to lack of regex match for nightly's of PHP (currently 8.0.0).

Spomky commented:
Hi @Kalloritis,

Thank you for this PR. This is something I was considering.
I have 2 comments:

* The [composer.json](https://github.com/web-token/jwt-framework/blob/v2.1/src/Component/Core/composer.json#L23) file from the "core" sub-package is missing
* Considering the way PHP evolves, it appears to me that `>=7.2` can be used instead of `^ 7.2|8.0`

Would you mind to update your PR?
Many thanks.

Kalloritis committed:
Update composer.json

psr/http-client 1.0.0 does not support php 8 (^7.0) and is latest branch
spomky-labs/aes-key-wrap 5.0.0 does not support php 8 (^7.1) and is latest branch
bjeavons/zxcvbn-php 0.4.0 does not support php 8 (^7.0) and latest branch (1.1.0) only supports up to ^7.1
spomky-labs/base64url v2.0.1/v2.0.0 does not support php 8 (^7.1) but latests branch (v2.x-dev after v2.0.1) suports PHP 8

Kalloritis commented:
PR should be updated- I noted several other upstream packages that would need to be updated for PHP 8 acceptance in their own composer.json files as well for review.

Spomky commented:
Thank you!
